ui/components/src/ui-components/tree/controlled/component/ControlledTree.tsx

```diff
@@ -26,12 +26,12 @@ import { RenderedItemsRange, TreeRenderer, TreeRendererProps } from "./TreeRende
  * @public
  */
 export interface ControlledTreeProps extends CommonProps {
-  /** Flat list of nodes to be rendered in tree. */
-  visibleNodes: VisibleTreeNodes;
+  /** Model of the tree to display. */
+  model: TreeModel;
   /** Node loader used to load root nodes and placeholder nodes. */
   nodeLoader: ITreeNodeLoader;
   /** Tree events handler. */
-  treeEvents: TreeEvents;
+  eventsHandler: TreeEvents;
   /** Mode of nodes' selection in tree. */
```

Comment:
There is nothing wrong with the name `eventsHandler` per se, but it is more usual to use `eventHandler`. There are already some places that pass this prop like this: `eventsHandler={eventHandler}`.

Reply:
There are also some others where the prop is assigned like `eventsHandler={eventsHandler}`... I personally like it in plural better, because the handler may handle multiple events, not just one.

Reply:
The amount of search results for "EventHandler" and "EventsHandler" on Google differ by three orders of magnitude. Also [event loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/EventLoop#event_loop) is a widely-known concept that deals with multiple kinds of events.

Reply:
> The amount of search results for "EventHandler" and "EventsHandler" on Google differ by three orders of magnitude.

Top 3 results for `EventHandler` are for a case where the handler is handling only one event.

> Also event loop is a widely-known concept that deals with multiple kinds of events.

Yes, but in case of a loop it also makes sense, because each iteration of the loop is handling one event.

Honestly, I don't care which one we choose, but I don't think it's worth spending time removing "s" which is already there.